src/view/actions/createMediaSession.jsx

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import form from "../forms/form";
+import instancePicker from "../forms/instancePicker";
+import conditional from "../forms/conditional";
+import textField from "../forms/textField";
+import dataElement from "../forms/dataElement";
+import renderForm from "../forms/renderForm";
+import codeField from "../forms/codeField";
+
+const getInitialValues = ({ initInfo }) => {
+  const {
+    instanceName = initInfo.extensionSettings.instances[0].name,
+    xdm = "",
+    playerId = "",
+    onBeforeMediaEvent = ""
+  } = initInfo.settings || {};
+
+  return {
+    instanceName,
+    xdm,
+    playerId,
+    onBeforeMediaEvent
+  };
+};
+
+const getSettings = ({ values }) => {
+  const settings = {
+    instanceName: values.instanceName
+  };
+  if (values.xdm) {
+    settings.xdm = values.xdm;
+  }
+  if (values.playerId) {
+    settings.playerId = values.playerId;
+  }
+  if (values.onBeforeMediaEvent) {
+    settings.onBeforeMediaEvent = values.onBeforeMediaEvent;
+  }
+
+  return settings;
+};
+
+const createSessionEventForm = form(
+  {
+    getInitialValues,
+    getSettings
```

Comment:
These should be `wrapGetInitialValues` and `wrapGetSettings`. 

`src/view/evaluateRulesets.jsx` is a simple example:

```js
const wrapGetInitialValues = getInitialValues => ({ initInfo }) => {
  const { personalization = {}, ...otherSettings } = initInfo.settings || {};
  return getInitialValues({
    initInfo: {
      ...initInfo,
      settings: { ...personalization, ...otherSettings }
    }
  });
};